src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -339,6 +339,19 @@ protected void cleanPayeeACHAccountExtractDetail(PayeeACHAccountExtractDetail de
                 LOG.error(logMessageStarter + " is not numeric after cleaning");
             }
         }
+
+        detail.setBankName(extractBankName(detail));
+    }
+
+    private String extractBankName(PayeeACHAccountExtractDetail detail) {
+        String bankName = ObjectUtils.isNull(detail) ? StringUtils.EMPTY : StringUtils.trim(detail.getBankName());
+
+        if (StringUtils.length(bankName) > CUPdpConstants.PAYEE_ACH_ACCOUNT_MAX_BANK_NAME_LENGTH) {
+            LOG.info("extractBankName truncating bank name to " + CUPdpConstants.PAYEE_ACH_ACCOUNT_MAX_BANK_NAME_LENGTH + " characters");
+            bankName = bankName.substring(0, CUPdpConstants.PAYEE_ACH_ACCOUNT_MAX_BANK_NAME_LENGTH);

```

Comment:
Does the bank name need to be trimmed a second time after being truncated? There might be cases where the 40th character is a space between words.

Reply:
This should be fixed now